src/app/model/ChessBoard.ts

```diff
@@ -0,0 +1,21 @@
+import { PiecePositions } from './PiecesPositions';
+import { Piece } from './Piece';
+import { SquareWithPiece, Square } from './Types';
+import { PieceMoves } from './PieceMoves';
+
+export class ChessBoard implements PieceMoves, PiecePositions {
+  constructor(private boardWithPieces: SquareWithPiece) {}
+
+  onPositionPiece(square: Square): Piece | undefined {
+    return this.boardWithPieces[`${square.column}${square.row}`] ? this.boardWithPieces[`${square.column}${square.row}`] : undefined;
```

Comment:
To nie ma sensu. Tzn ma ale niepotrzebnie komplikuje :) 
`return this.boardWithPieces[`${square.column}${square.row}`] ? this.boardWithPieces[`${square.column}${square.row}`] : undefined;`
jest równoważne:
`return this.boardWithPieces[`${square.column}${square.row}`]`
To po prostu zwraca undefined lda propertiesa, którego nie ma. Dlatego też nie używamy null, bo zapis jest prostszy. Z nullem musiałbyć zrobić tak jak napisałeś.

Reply:
racja racja! :-)